packages/SwingSet/src/lib/kmarshal.js

```diff
@@ -0,0 +1,55 @@
+import { Far, makeMarshal } from '@endo/marshal';
+
+// Simple wrapper for serializing and unserializing marshalled values inside the
+// kernel, where we don't actually want to use clists nor actually allocate real
+// objects, but instead to stay entirely within the domain of krefs.  This is
+// used to enable syntactic manipulation of serialized values while remaining
+// agnostic about the internal details of the serialization encoding.
+
+const promiseMap = new WeakMap();
+
+export const kslot = (kref, iface) => {
+  if (iface && iface.startsWith('Alleged: ')) {
+    // Encoder prepends "Alleged: " to iface string, but the decoder doesn't strip it
+    // Unclear whether it's the decoder or me who is wrong
+    iface = iface.slice(9);
+  }
+  if (
+    kref.startsWith('p') ||
```

Comment:
There's a weakness in how the functions in this module are named (i.e., they're mostly `kshortthing`), because there _are_ uses of these functions (in test code and comms code) that do involve vrefs and comms refs.  I'm very open to better names (in particular, the main reason I haven't already ditched `kslot`, a name that's arguably terrible in multiple ways, is that it is very weildy, more so than any of the competing alternatives I've considered).

However, I don't think the particular paranoia of explicitly checking for `p+` and `p-` buys us anything, since the `p` check is entirely sufficient for the current ref encodings and if we ever change or extend them we'll have to overhaul this function anyway.